sale_order_pricelist_commitment_date/models/sale_order.py

```diff
@@ -9,8 +9,24 @@ class SaleOrder(models.Model):
 
     @api.onchange("commitment_date")
     def onchange_commitment_date(self):
-        if self.commitment_date:
+        if self.commitment_date and self.pricelist_id and self.partner_id:
             for line in self.order_line:
-                line.with_context(
-                    commitment_date=self.commitment_date
-                ).product_id_change()
+                product = line.product_id.with_context(
+                    lang=self.partner_id.lang,
+                    partner=self.partner_id,
+                    quantity=line.product_uom_qty,
+                    date=self.date_order,
+                    pricelist=self.pricelist_id.id,
+                    uom=line.product_uom.id,
+                    commitment_date=self.commitment_date,
```

Comment:
```suggestion
                    date=self.commitment_date,
                    pricelist=self.pricelist_id.id,
                    uom=line.product_uom.id,
                    fiscal_position=self.env.context.get("fiscal_position"),
```

How about making it like this, and removing the adjustment on `_compute_price_rule()`?